### PR TITLE
Add linux cpu test for 3.12

### PR DESCRIPTION
**Pull request by albanD**
cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @aakhundov @kadeng

### CODE REVIEW

File: .github/workflows/pull.yml [hunk 1]
```diff
@@ -175,6 +175,32 @@ jobs:
...
+      docker-image-name: pytorch-linux-focal-py3.12-clang10
+      test-matrix: |
+        { include: [
+          { config: "default", shard: 1, num_shards: 3, runner: "linux.2xlarge" },
+          { config: "default", shard: 2, num_shards: 3, runner: "linux.2xlarge" },
+          { config: "default", shard: 3, num_shards: 3, runner: "linux.2xlarge" },
+          { config: "crossref", shard: 1, num_shards: 2, runner: "linux.2xlarge" },
+          { config: "crossref", shard: 2, num_shards: 2, runner: "linux.2xlarge" },
+          { config: "dynamo", shard: 1, num_shards: 2, runner: "linux.2xlarge" },
+          { config: "dynamo", shard: 2, num_shards: 2, runner: "linux.2xlarge" },
```
Comment (malfet) [inline .github/workflows/pull.yml:192]: Can we really run `dynamo` tests in 3.12?

**Review by malfet — APPROVED** [review]
LGTM, but I wonder whenever one adds a config to the `pull.yaml`, they should consider removing one as well (i.e. why keep 3.11 config for non-inductor tests if we are already testing 3.12 one)

File: .github/workflows/pull.yml (hunk 1 above)
Comment (albanD) [reply .github/workflows/pull.yml:192]: No, I was curious what was going to be the failure and wanted to make sure it failed nicely.

**Comment by albanD** [comment]
The main failure for non-dynamo tests is due to missing lapack during compile

**Comment by albanD** [comment]
Closing in favor of https://github.com/pytorch/pytorch/pull/117853

**Pull request closed without merging**